Tools/CISupport/build-webkit-org/steps.py

```diff
@@ -1565,6 +1565,201 @@ def getResultSummary(self):
+        kwargs['timeout'] = 2 * 60 * 60
+        super(ScanBuildSmartPointer, self).__init__(**kwargs)
+        self.commandFailed = False
+
+    def start(self):
+        self.command = '/bin/rm -rf {output_dir}; '.\
+            format(output_dir=os.path.join(self.getProperty('builddir'), 'build/scan-build-output'))
+        self.command += 'Tools/Scripts/build-and-analyze --output-dir {output_dir} '.\
+            format(output_dir=os.path.join(self.getProperty('builddir'), 'build/scan-build-output'))
+        self.command += '--only-smart-pointers --analyzer-path={analyzer_path} '.format(analyzer_path=os.path.join(self.getProperty('builddir'), 'llvm-project/build/bin/clang'))
+        self.command += '--scan-build-path=../llvm-project/clang/tools/scan-build/bin/scan-build '
+        self.command += '--sdkroot=macosx '
+        self.command += '2>&1 | python3 Tools/Scripts/filter-static-analyzer'
+
+        self.log_observer = logobserver.BufferLogObserver()
+        self.addLogObserver('stdio', self.log_observer)
+        return shell.ShellCommand.start(self)
+
+    def commandComplete(self, cmd):
+        shell.ShellCommand.commandComplete(self, cmd)
+
+        log_text = self.log_observer.getStdout()
+        index = log_text.rfind('Total issue count:')
+        if index != -1:
+            total_issue_count_regex = re.compile(r'^Total issue count: (\d+)$', re.MULTILINE)
+            match = total_issue_count_regex.search(log_text[index:])
+            if match:
+                self.bugs += int(match.group(1))
+
+        f_index = log_text.rfind('ANALYZE SUCCEEDED')
+        if f_index == -1:
+            self.analyzeFailed = True
+
+        steps_to_add = [
+            GenerateS3URL(
+                f"{self.getProperty('fullPlatform')}-{self.getProperty('architecture')}-{self.getProperty('configuration')}-{self.name}",
+                extension='txt',
+                content_type='text/plain',
+            ), UploadFileToS3(
+                'build-log.txt',
+                links={self.name: 'Full build log'},
+                content_type='text/plain',
+            )
+        ]
+        self.build.addStepsAfterCurrentStep(steps_to_add)
+
+    def evaluateCommand(self, cmd):
+        if cmd.rc != 0:
+            self.commandFailed = True
+            return FAILURE
+        if self.analyzeFailed:
+            return FAILURE
+
+        return SUCCESS
+
+    def getResultSummary(self):
+        status = ''
+        if self.analyzeFailed:
+            status += 'ANALYZE FAILED: '
+        status += self.name
+
+        if self.commandFailed:
+            status += ' failed to build, and'
+        status += ' found {} issues'.format(self.bugs)
+
+        if self.results != SUCCESS:
+            status += ' ({})'.format(Results[self.results])
+        else:
+            self.build.addStepsAfterLastStep([UpdateSmartPointerBaseline()])
+
+        return {u'step': status}
+
+
+class ParseStaticAnalyzerResults(shell.ShellCommand):
```

Comment:
Can we use a `shell.ShellCommandNewStyle`, so we don't have to go back and update this class in the future when we update buildbot versions?